core/src/main/java/org/elasticsearch/search/suggest/term/TermSuggestionBuilder.java

```diff
@@ -233,25 +347,128 @@ public String getWriteableName() {
 
     @Override
     public void doWriteTo(StreamOutput out) throws IOException {
-        // NORELEASE
-        throw new UnsupportedOperationException();
+        out.writeOptionalVInt(suggestMode != null ? suggestMode.ordinal() : null);
+        out.writeOptionalFloat(accuracy);
+        out.writeOptionalVInt(sort != null ? sort.ordinal() : null);
+        out.writeOptionalVInt(stringDistance != null ? stringDistance.ordinal() : null);
```

Comment:
Can you put the read/write logic for the enums into the respective class? That way it's easier to see how these enums are sent over the wire.